src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,35 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
 public class SalaryInfo {
+    private static final SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder(
+                "Report for period " + dateFrom + " - " + dateTo);
+        String[] temp;
+        int sum = 0;
+        for (int i = 0; i < data.length; i++) {
+            for (String oneData: data) {
+                temp = oneData.split(" ");
+                try {
+                    if (names[i].equals(temp[1])
+                            && ((sdf.parse(temp[0]).compareTo(sdf.parse(dateFrom))) >= 0
+                            && (sdf.parse(temp[0]).compareTo(sdf.parse(dateTo))) <= 0)) {
+                        sum = sum + Integer.parseInt(temp[2]) * Integer.parseInt(temp[3]);
+                    }
+                } catch (ParseException e) {
+                    throw new RuntimeException(e);
```

Comment:
If you want to throw some expection it is better to throw specific one, not generic. And always define exception message